src/odemis/odemisd/modelgen.py

```diff
@@ -436,12 +442,14 @@ def _make_args(self, name):
             init["model"] = self.ast
 
         # create recursively the children
-        if "children" in init:
+        if "children" in init or "dependencies" in init:
```

Comment:
Just create a separate if block for "dependencies" in init, which raises a different SemanticError, with the accurate error message.